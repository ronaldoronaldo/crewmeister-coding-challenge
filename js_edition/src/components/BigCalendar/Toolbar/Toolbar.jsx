import React from "react";
import moment from "moment";
import { ButtonsWrapper, ToolbarContainer, ToolbarButton, Title } from "./Toolbar.style.jsx";

export default function Toolbar(toolbar) {
  const goToDayView = () => {
    toolbar.onView('day');
  }
  const goToWeekView = () => {
    toolbar.onView('week');
  }
  const goToMonthView = () => {
    toolbar.onView('month');
  }

  const goToBack = () => { toolbar.onNavigate('PREV'); };
  const goToNext = () => { toolbar.onNavigate('NEXT'); };
  const goToCurrent = () => { toolbar.onNavigate('TODAY'); };

  const renderTitle = () => {
    const date = moment(toolbar.date);
    return (
      <Title><b>{date.format('MMMM')}</b> {date.format('YYYY')}</Title>
    );
  };

  return (
    <ToolbarContainer>
      <ButtonsWrapper>
        <ToolbarButton onClick={goToCurrent}>Today</ToolbarButton>
        <ToolbarButton onClick={goToBack}>Back</ToolbarButton>
        <ToolbarButton onClick={goToNext}>Next</ToolbarButton>
      </ButtonsWrapper>
      {renderTitle()}
      <ButtonsWrapper>
        <ToolbarButton onClick={goToDayView}>Day</ToolbarButton>
        <ToolbarButton onClick={goToWeekView}>Week</ToolbarButton>
        <ToolbarButton onClick={goToMonthView}>Month</ToolbarButton>
      </ButtonsWrapper>
    </ToolbarContainer>
  );
}
