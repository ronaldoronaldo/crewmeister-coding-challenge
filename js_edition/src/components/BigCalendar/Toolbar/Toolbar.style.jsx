import styled from "styled-components";
import {colors, fontWeights, radius, screens, spacing, text, transitions} from "../../../config/styleGuide";

export const ToolbarStyle = styled.div`

`;
