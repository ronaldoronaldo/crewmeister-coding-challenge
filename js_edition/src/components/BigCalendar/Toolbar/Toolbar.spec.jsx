import React from "react";
import { render } from "@testing-library/react";
import moment from "moment";

import Toolbar from "./Toolbar";

describe("<Event />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Toolbar />);
    expect(container).toMatchSnapshot();
  });
});
