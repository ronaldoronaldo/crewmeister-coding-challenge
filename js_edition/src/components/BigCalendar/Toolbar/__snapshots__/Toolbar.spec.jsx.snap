// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> should match snapshot 1`] = `
<div>
  <div
    class="sc-AxhCb kCcHTH"
  >
    <div
      class="sc-AxiKw emcivl"
    >
      <button
        class="sc-AxirZ sc-AxhUy dTzrWs"
      >
        Today
      </button>
      <button
        class="sc-AxirZ sc-AxhUy dTzrWs"
      >
        Back
      </button>
      <button
        class="sc-AxirZ sc-AxhUy dTzrWs"
      >
        Next
      </button>
    </div>
    <span
      class="sc-AxgMl gCVfLD"
    >
      <b>
        July
      </b>
       
      2020
    </span>
    <div
      class="sc-AxiKw emcivl"
    >
      <button
        class="sc-AxirZ sc-AxhUy dTzrWs"
      >
        Day
      </button>
      <button
        class="sc-AxirZ sc-AxhUy dTzrWs"
      >
        Week
      </button>
      <button
        class="sc-AxirZ sc-AxhUy dTzrWs"
      >
        Month
      </button>
    </div>
  </div>
</div>
`;
